### PR TITLE
Added Solution #0623

### PR DESCRIPTION
**Pull request by The-Divyansh**
Done all the changes suggested with proper formatting and LaTex

## Change Summary

Provide summary of changes with issue number if any.

## Checklist

**If you haven't fulfilled the below requirements, your PR won't be reviewed.**

### General 

- [x] Make sure you've read [CONTRIBUTING.md](https://github.com/wingkwong/leetcode-the-hard-way/blob/main/CONTRIBUTING.md) 
- [x] Make sure you've followed the template format
- [x] Make sure you've started the app locally and verified all the content and all links (if applicable) must be accessible correctly
- [x] Make sure you've applied LaTex for variables, formulas and time / space complexity instead of using backticks
- [x] Make sure you've formatted your code well (Code Block is rendered using `<Tabs/>`)
      Below is a C++ example. You can include multiple langs using `<TabItem/>` if necessary
      
````
<Tabs>

<TabItem value="cpp" label="C++">
<SolutionAuthor name="@YOUR_ALIAS"/>

```cpp
// Your code goes here
// Prefer K&R Coding style
```

</TabItem>

</Tabs>
````
- [x] Make sure you've explained your code well (write comments above each line - not inline)
- [x] Make sure you've written your explanation well and it is easy to understand for beginners
- [x] Make sure you've included Time & Space Complexity Analysis
- [x] Make sure you've included a meaningful approach name for your solution. Don't leave it blank. e.g. `## Approach 1: Two Pointers`.

### For Tutorial Articles

- [ ] Make sure you've explained your topic using 1 - 3 LC problems rather than using external problems. Solutions must be taken with explanations as well.
- [ ] Make sure you've provided suggested problems at the end using the given format. See [here](https://raw.githubusercontent.com/wingkwong/leetcode-the-hard-way/main/tutorials/math/number-theory/binary-exponentiation.md) as an example. If the target solution is not available, leave `solutionLink` blank.
- [ ] Make sure you've given credits / references if you use external resources. For an image, give credit under it. Otherwise, add a new section called References at the end (after Suggested Problems).

### CODE REVIEW

**Comment by The-Divyansh** [comment]
@vigneshshiv @wingkwong  please check

**Comment by wingkwong** [comment]
Just for record. Previous comments logged in #207 #212

File: solutions/0600-0699/0623-add-one-row-to-tree-medium.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    }
+
+    TreeNode* addOneRow(TreeNode* root, int val, int depth) {
+        // since root is changed in this case, handle this case seperately
+        if(depth==1)
+        {
+            TreeNode* newNode = new TreeNode(val);
+            newNode->left = root;
+            return newNode;
+        }
```
Comment (wingkwong) [inline solutions/0600-0699/0623-add-one-row-to-tree-medium.md:124]: follow this coding style for all. i.e. the space and bracket position

```cpp
if (depth == 1) {
    TreeNode* newNode = new TreeNode(val);
    newNode->left = root;
    return newNode;
}
```

File: solutions/0600-0699/0623-add-one-row-to-tree-medium.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+ */
+class Solution {
+public:
+    void insert(TreeNode* node, int currDepth, int val, int depth)
+    {
+        if(!node) return;
+
+        //if we reach the node where we need to create two nodes
+        if(currDepth == depth-1)
+        {
```
Comment (wingkwong) [inline solutions/0600-0699/0623-add-one-row-to-tree-medium.md:95]: space and bracket. see below comment

File: solutions/0600-0699/0623-add-one-row-to-tree-medium.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            node->left = l;
+            node->right = r;
+
+            // return since we do not need to further deep in the tree
+            return;
+        }
+
+        // if we do not reach desired depth, go deep in the tree
+        insert(node->left, currDepth+1, val, depth);
+        insert(node->right, currDepth+1, val, depth);
```
Comment (wingkwong) [inline solutions/0600-0699/0623-add-one-row-to-tree-medium.md:114]: space.

```
currDepth + 1
```

File: solutions/0600-0699/0623-add-one-row-to-tree-medium.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    TreeNode* addOneRow(TreeNode* root, int val, int depth) {
+        // since root is changed in this case, handle this case seperately
+        if(depth==1)
+        {
+            TreeNode* newNode = new TreeNode(val);
+            newNode->left = root;
+            return newNode;
+        }
+
+        // recursive function for insertion of new nodes (similar to dfs)
```
Comment (wingkwong) [inline solutions/0600-0699/0623-add-one-row-to-tree-medium.md:126]: `(similar to dfs)` can be deleted.

File: solutions/0600-0699/0623-add-one-row-to-tree-medium.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+## Approach 1: Recursion (DFS)
+
+This approach uses a simple DFS approach. The recursive function $insert(node, currDepth, val, depth)$ can be used to recursively reach a node where we need to insert two nodes.
+
+The parameters used in the function are:
+- node -> the current node on which DFS is called upon
+- currDepth -> the depth of current node $node$
+- val -> value of nodes to be inserted
+- depth -> the depth at which nodes are to be inserted
```
Comment (wingkwong) [inline solutions/0600-0699/0623-add-one-row-to-tree-medium.md:59]: don't use `->` use `:`

**Review by wingkwong — CHANGES_REQUESTED** [review]
Don't close your PR. Push the changes and this PR will be updated automatically.

**Comment by The-Divyansh** [comment]
@wingkwong @vigneshshiv updated explanation and formatted the code. Please check

**Comment by wingkwong** [comment]
I cleaned it up. Don't wanna waste time. This is what I'm expecting. https://github.com/wingkwong/leetcode-the-hard-way/pull/224/commits/cb7036df0d4c3a9efe87125becc49d9efabf0bc5.

**Pull request merged**